trading_venue_open.py

```diff
@@ -1,79 +1,68 @@
 import os
 from dotenv import load_dotenv
 import json
+from lemon import api
 
+from helpers import EmailSenderSendgrid
 
-from helpers import RequestHandler, EmailSenderSendgrid
 
+class TradingVenue():
+    load_dotenv()
+
+    def __init__(self):
+        self.client = api.create(
+            market_data_api_token=os.getenv("DATA_API_KEY"),
+            trading_api_token=os.getenv("TRADING_API_KEY"),
+            env="paper"
+    )
 
-class TradingVenue(RequestHandler):
-    
     def send_out_email(self):
-        load_dotenv()
-        endpoint = f'positions/'
-        response = self.get_data_trading(endpoint)
+        response = self.client.trading.positions.get()
         email_text = json.dumps(response)
         subject = "Your positions at market close"
         EmailSenderSendgrid(email_text, subject)
 
-    def check_if_open(self):
-        """
-        helper function to check if the trading venue is currently open
-        :return: {Boolean} is returned that tells whether trading venues is open or closed
-        """
-        load_dotenv()
-        mic = os.getenv("MIC")
-        endpoint = f'venues/?mic={mic}'
-        response = self.get_data_data(endpoint)
-        is_open = response['results'][0].get('is_open', 'It was not possible to retrieve the is_open attribute')
-        print(is_open)
-        return is_open
-
-    def activate_order(self, order_id):
-        """
-        helper function to activate the order once it was placed
-        :param order_id: the order ID of the order that is to be activated
-        """
-        load_dotenv()
-        endpoint = f'orders/{order_id}/activate/'
-        response = self.post_data(endpoint, {})
-        print(response)
-
-    def place_order(self):
+    def place_order_with_email(self):
         """
         main function that places the specified order
         """
-        load_dotenv()
-        if not self.check_if_open():
-            email_text = "Hey there. You tried to place an order with the lemon.markets API, but the market is currently closed. Please try again later."
-            subject = "The market is currently closed"
+        mic = os.getenv("MIC")
+        venue = self.client.market_data.venues.get(mic=mic).results[0]
+        if not venue.is_open:
+            email_text = "Hey there. You tried to place an order with the lemon.markets API, but the market is " \
+                         "currently closed. Please try again later."
+            subject = "Trade failed: the market is currently closed."
             EmailSenderSendgrid(email_text, subject)
             return
 
         # send notification email if trading venue is closed
 
         try:
-            mic = os.getenv("MIC")
-
             order_details = {
                 "isin": "DE0008232125",  # ISIN of Lufthansa
-                "expires_at": "p7d",  # specify your timestamp
+                "expires_at": 7,  # specify your timestamp
```

Comment:
Would opt to remove this dict structure here - made sense with the previous implementation because we could feed the dictionary directly, but imo calling order_details.get("") every time isn't super readable.

Instead, just define quantity & isin as variables, perform the minimum order price check and directly define the remaining variables within the orders.create() function.